### ISSUE TITLE
os-frr: BGP authentication not always working

### ISSUE THREAD

**Issue by joachimtingvold** [opnsense/plugins#3508]
**Important notices**
Before you add a new report, we ask you kindly to acknowledge the following:

- [x] I have read the contributing guide lines at https://github.com/opnsense/plugins/blob/master/CONTRIBUTING.md
- [x] I have searched the existing issues, open and closed, and I'm convinced that mine is new.
- [x] The title contains the plugin to which this issue belongs

**Describe the bug**
Adding MD5 authentication to a peer often results in "can't connect to [peer] fd 23 : No such file or directory".

**To Reproduce**
1. Add IPv4 and IPv6 BGP peer
2. Add password in the "BGP MD5 Password" field
3. Add the local IP (IPv4 or IPv6) in the "Local Initiater IP" field
4. Save the configuration
5. Apply the configuration (via the small circle button)
6. Reload/restart FRR/BGP 
7. Check that the configuration looks sane (see excerpt below), and that it matches the local `/usr/local/etc/frr/bgpd.conf` file
8. Watch the log (Routing -> Diagnostic -> Log), and that it can't connect to the neighbor

**Expected behavior**
Peer should connect with MD5 authentication.

**Screenshots**
![Screen-Shot-2023-07-23-14-53-08](https://github.com/opnsense/plugins/assets/359918/811fb347-e5d1-4a28-bb98-8def2e58e343)
![Screen-Shot-2023-07-23-14-53-19](https://github.com/opnsense/plugins/assets/359918/9030a8c9-de65-423d-8cd4-f3e17ccdaa3f)

**Relevant log files**
```
2023-07-23T15:04:29 | Informational | bgpd | can't connect to <ipv4-peer> fd 25 : No such file or directory
2023-07-23T15:04:29 | Informational | bgpd | can't connect to <ipv6-peer> fd 25 : No such file or directory
```

**Additional context**
```
!
frr version 7.5.1
frr defaults traditional
hostname <snip>
log syslog informational
!
router bgp 64601
 bgp router-id <snip>
 no bgp ebgp-requires-policy
 no bgp default ipv4-unicast
 bgp graceful-restart
 neighbor <v4-peer> remote-as 64601
 neighbor <v4-peer> bfd
 neighbor <v4-peer> password bV...xN
 neighbor <v4-peer> update-source vlan0601
 neighbor <v6-peer> remote-as 64601
 neighbor <v6-peer> bfd
 neighbor <v6-peer> password bV...xN
 neighbor <v6-peer> interface vlan0601
 !
 address-family ipv4 unicast
  redistribute kernel
  redistribute connected
  neighbor <v4-peer> activate
  neighbor <v4-peer> next-hop-self
  neighbor <v4-peer> prefix-list any_ipv4 in
  neighbor <v4-peer> prefix-list any_ipv4 out
 exit-address-family
 !
 address-family ipv6 unicast
  redistribute kernel
  redistribute connected
  neighbor <v6-peer> next-hop-self
  neighbor <v6-peer> prefix-list any_ipv6 in
  neighbor <v6-peer> prefix-list any_ipv6 out
 exit-address-family
!
ip prefix-list any_ipv4 seq 300 deny 0.0.0.0/0
ip prefix-list any_ipv4 seq 310 permit 0.0.0.0/0 le 32
ip prefix-list default_route_ipv4 seq 100 permit 0.0.0.0/0
!
ipv6 prefix-list any_ipv6 seq 400 deny ::/0
ipv6 prefix-list any_ipv6 seq 410 permit ::/0 le 128
ipv6 prefix-list default_route_ipv6 seq 200 permit ::/0
!
line vty
!
bfd
 peer <v4-peer>
 !
 peer <v6-peer>
 !
!
```

**Environment**
OPNsense 23.1.11-amd64
Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz (20 cores, 40 threads)
Network Intel X520-DA2

**Comment by joachimtingvold** [comment]
Continues the discussion from end of #1972. I initially thought this was relating to only IPv6 peers, as I sometimes get it to work with IPv4 peers (by reloading/removing/adding configuration frantically until it suddenly works, even if it's the exact same configuration being added). However, after I learned that the original PR #2800 (that only supported IPv4 peers) was updated to support IPv6 peers (via PR #3345), I've attempted to get this working, where I now can see that the "can't connect to <peer> fd 25 : No such file or directory" happens randomly for both IPv4 and IPv6 peers.

**Comment by joachimtingvold** [comment]
If I remove the contents of the "BGP MD5 Password" and "Local Initiater IP" fields in my example above, then save + apply + reload, the neighbor comes up without any issues (without changing any other configuration aspects).

**Comment by joachimtingvold** [comment]
Disabling BFD (both globally, and on the peer) does not change the outcome.

**Comment by farthinder** [comment]
For what it is worth I am running in to this problem as well "can't connect to 192.168.x.x fd 23 : No such file or directory"

I´ve set the "Local Initiater IP" and "Update-Source Interface".

Not sure if it is related but the IP and Interface is a WireGuard interface

**Comment by Andre15711** [comment]
I ran into the same issue...
Did someone found a solution for this?

**Comment by mikebutash** [comment]
I was fighting with this myself in turning up a peering, and was related to the Local Initiator IP not being set for the security association (verify with "egrep '^\[|^(src|dst|protocol|spi|aalgo)=' /usr/local/etc/frr/sa_policies.conf
"). Once I set this and both sides register both src/dst, it immediately came up and stopped with the errors.

I did try using a more complex 16 char pass with special characters after getting working that failed to then work, assuming one side or the other (cloudflare wan) is failing to escape the characters. That might be causing some of your issues too, or was in my case, and had to stick with a more simple long alpha-numeric pass without special characters.
